src/repr/src/row/encoding2.rs

```diff
@@ -1149,8 +1149,18 @@ impl RowColumnarDecoder {
 
         // For performance reasons we downcast just a single time.
         let mut decoders = Vec::with_capacity(desc_columns.len());
-        for (col_array, col_type) in inner_columns.iter().zip(desc_columns.iter()) {
-            let decoder = array_to_decoder(col_array, &col_type.scalar_type)?;
+        let mut itoa = itoa::Buffer::new();
+
+        // The columns of the `StructArray` are named with their column index.
+        for (col_idx, col_type) in desc_columns.iter().enumerate() {
+            let field_name = itoa.format(col_idx);
```

Comment:
I'm tempted to suggest just going with `format!` because:
- Can add a prefix. Deep in my heart I worry we're going to have problems with columns named `10` instead of something more obviously namey like `f10` or something.
- Hard to imagine it matters outside the inner loop, and it's nice to save a dep.

Reply:
Sounds good! Removed the `itoa` dep